src/DynamoCoreWpf/UI/Converters.cs

```diff
@@ -1686,4 +1734,236 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    // It's used for ClassDetails and ClassObject itself. ClassDetails should be not focusable,
+    // in contrast to ClassObject.
+    // Also decides, should be category underlined or not.
+    public class ElementTypeToBoolConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value is NodeSearchElementViewModel)
+                return false;
+            if (value is RootNodeCategoryViewModel)
+            {
+                var rootElement = value as RootNodeCategoryViewModel;
+                return !rootElement.SubCategories.OfType<ClassesNodeCategoryViewModel>().Any();
+            }
+            if (value is NodeCategoryViewModel)
+                return true;
+
+            return false;
+        }
+
+        public object ConvertBack(
+            object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    // This converter is used to change color of output parameters for custom node.
+    public class NodeTypeToColorConverter : IValueConverter
+    {
+        public SolidColorBrush TrueBrush { get; set; }
+        public SolidColorBrush FalseBrush { get; set; }
+
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value is CustomNodeSearchElementViewModel)
+                return TrueBrush;
+            return FalseBrush;
+        }
+
+        public object ConvertBack(
+            object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public class RootElementVMToBoolConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return (value is RootNodeCategoryViewModel);
+        }
+
+        public object ConvertBack(
+            object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+    }
+    // Used in addons treeview. Element, that is just under root shouldn't have dotted line at the left side.
+    public class HasParentRootElement : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value is RootNodeCategoryViewModel) return true;
+            if (value is NodeCategoryViewModel)
+            {
+                // TODO(Vladimir): take a look.
+                return false; //(value as BrowserInternalElementViewModel).CastedModel.Parent is BrowserRootElement;
+            }
+            else return false;
```

Comment:
If you remove the `TODO` comment (which I think is obsolete now), you can simplify this as:

```csharp
public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
{
    return (value is RootNodeCategoryViewModel);
}
```

If that is the case, `HasParentRootElement` will be redundant, because it will be the same as `RootElementVMToBoolConverter` above. In such case, please remove it and update its caller to use `RootElementVMToBoolConverter` instead.